src/Stubs.cs

```diff
@@ -1,23 +1,38 @@
 using SpacetimeDB.ClientApi;
 
+using System;
+
 namespace SpacetimeDB
 {
     public interface IReducerArgsBase : BSATN.IStructuralReadWrite
     {
         string ReducerName { get; }
     }
 
-    public abstract class ReducerEventBase
+    public abstract class DbContext<DbView, ReducerView> : DbContext<DbView>
+        where DbView : class, new()
+        where ReducerView : class, new()
+    {
+        public ReducerView Reducers;
+
+        public DbContext(DbView db, ReducerView reducers) : base(db) => Reducers = reducers;
+    }
+
+    public interface IEventContext {
+        bool InvokeHandler();
+    }
+
+    public abstract class EventContextBase<RemoteTables, RemoteReducers> : DbContext<RemoteTables, RemoteReducers>, IEventContext
+        where RemoteTables : class, new()
+        where RemoteReducers : class, new()

```

Comment:
This bound doesn't work anymore. CI would probably catch it, but it doesn't get that far.

This class doesn't need the `new()` constraints though, or, for that matter, any constraints for those generics, since it accepts them from outside.

Reply:
(fixed this one myself)